app/forms/declaration/GoodsLocation.scala

```diff
@@ -31,7 +31,17 @@ case class GoodsLocation(
   addressLine: Option[String],
   postCode: Option[String],
   city: Option[String]
-)
+) {
+
+  def extractLocationCode: String =
+    retrieveCountryFromName(country).countryCode + typeOfLocation + qualifierOfIdentification + identificationOfLocation.getOrElse("")
+
+  def extractAddressValues: Seq[String] = {
+    val address = Seq(addressLine, city, postCode, Some(country)).flatten
+
+    if (address.length > 3) address else Seq.empty
```

Comment:
Why "3"?   Does this mean that if we don't have all four address elements then we return (show) nothing?

Reply:
Yes, frontend logic, moved to the view, but still there is the same if. I can wrap it in some value to describe this code. What do you think? Something like

`val isAddressContainsAllFields = if(address.length > 3)`

`if(isAddressContainsAllFields) address else Seq.empty`

Is it better? Do you have another idea

Reply:
Not that concerned how the logic is implemented, but if that logic is for the view, I'd move it there.

Reply:
So this is moved to the view

Reply:
Looks like it's still there?